crates/server/src/server/mod.rs

```diff
@@ -902,6 +909,18 @@ impl<R: ResponseHandler> ResponseHandler for ReportingResponseHandler<R> {
             );
         }
 
+        #[cfg(feature = "metrics")]
+        {
+            self.metrics.request_proto.increment(&self.protocol);
+            self.metrics
+                .request_operation
+                .increment(&self.request_header.op_code());
+            self.metrics.request_flags.increment(&self.request_header);
+
+            self.metrics.response_code.increment(&response_code);
+            self.metrics.response_flags.increment(&response_info)
+        }
```

Comment:
Suggest that these calls should be in a `ResponseHandlerMetrics::update()` (open to suggestions on the method name) method that takes `&response_info` and `&self`.

Reply:
Refactored.